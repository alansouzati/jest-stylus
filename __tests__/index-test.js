import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import path from 'path';

import jestStylus from '../';

Enzyme.configure({ adapter: new Adapter() });

test('react renders without style', () => {
  jestStylus(path.resolve(__dirname, 'index-test.styl'));
  const component = renderer.create(
    <div>test react no style</div>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('react renders style', () => {
  jestStylus(path.resolve(__dirname, 'index-test.styl'));
  const component = renderer.create(
    <div className='test test2'>test react with style</div>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('enzyme mounts without style', () => {
  jestStylus(path.resolve(__dirname, 'index-test.styl'));
  const component = mount(
    <div>test no style</div>
  );
  expect(component.getDOMNode()).toMatchSnapshot();
});

test('enzyme fails with invalid style', () => {
  jestStylus(path.resolve(__dirname, 'index-invalid.styl'));
  const component = mount(
    <div className='test'>test no style</div>
  );
  try {
    expect(component.getDOMNode()).toMatchSnapshot();
    fail('should not allow invalid style');
  } catch (e) {
    expect(e).toMatchSnapshot();
  }
});

test('enzyme mounts with style', () => {
  jestStylus(path.resolve(__dirname, 'index-test.styl'));
  const component = mount(
    <div className='test test2'><div className='test3'>children</div>test style</div>
  );
  expect(component.getDOMNode()).toMatchSnapshot();
});

test('enzyme mounts with styles', () => {
  jestStylus([path.resolve(__dirname, 'index-test.styl'), path.resolve(__dirname, 'index-test2.styl')]);
  const component = mount(
    <div className='test test2'>test styles</div>
  );
  expect(component.getDOMNode()).toMatchSnapshot();
});
