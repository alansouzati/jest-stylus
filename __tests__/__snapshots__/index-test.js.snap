// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme fails with invalid style 2`] = `
[ParseError: /Users/asouza/opensource-workspace/jest-stylus/__tests__/index-invalid.styl:3:2
   1| test
   2|   color #000
   3| }
-------^
   4| 

unexpected "}"
]
`;

exports[`enzyme mounts with style 1`] = `
.test {
  background: #00f;
}

.test3 {
  border: 1px solid #000;
}

@media screen and (min-width: 480px) {
  .test {
    font-size: 20px;
  }
}

<div
  class="test test2"
>
  <div
    class="test3"
  >
    children
  </div>
  test style
</div>
`;

exports[`enzyme mounts with styles 1`] = `
.test {
  background: #00f;
}

@media screen and (min-width: 480px) {
  .test {
    font-size: 20px;
  }
}

div.test2 {
  color: #000;
}

<div
  class="test test2"
>
  test styles
</div>
`;

exports[`enzyme mounts without style 1`] = `
<div>
  test no style
</div>
`;

exports[`react renders style 1`] = `
.test {
  background: #00f;
}

@media screen and (min-width: 480px) {
  .test {
    font-size: 20px;
  }
}

<div
  className="test test2"
>
  test react with style
</div>
`;

exports[`react renders without style 1`] = `
<div>
  test react no style
</div>
`;
